model/src/main/java/org/projectnessie/model/ContentKey.java

```diff
@@ -173,9 +179,17 @@ protected void check() {
     if (getElements().stream().mapToInt(String::length).sum() > MAX_LENGTH) {
       throw new IllegalStateException("Key too long, max allowed length: " + MAX_LENGTH);
     }
-    if (getElements().size() > MAX_ELEMENTS) {
+    if (getElementCount() > MAX_ELEMENTS) {
       throw new IllegalStateException(
           "Key too long, max allowed number of elements: " + MAX_ELEMENTS);
     }
   }
+
+  public ContentKey toParent() {
```

Comment:
Nit: `toParent` gives the impression that the ContentKey is being transformed into its parent. I think `getParent` is a bit more suitable.